File: servers/go/gin/src/api/v1/resources/users/schemas/login_schema.go
package schemas

import (
	"github.com/gin-gonic/gin"
	"github.com/tinadim/polyglot-rest-api/servers/go/gin/src/api/v1/lib/validation"
	"github.com/xeipuuv/gojsonschema"
)

func GetLoginSchema() gojsonschema.Schema {
	jsonSchema := gin.H{
		"type": "object",
		"properties": gin.H{
			"username": gin.H{
				"type":      "string",
				"minLength": 1,
			},
			"password": gin.H{
				"type":      "string",
				"minLength": 1,
			},
		},
		"required": [2]string{"password", "username"},
	}
	return validation.CreateBodySchema(jsonSchema)
}

File: servers/typescript/express/src/api/v1/resources/users/schemas/login-schema.ts
import { createBodyRequestSchema } from '@lib/schemas/request-schema';

export interface LoginBody {
    password: string;
    username: string;
}

export const loginSchema = createBodyRequestSchema<LoginBody>({
    properties: {
        password: {
            minLength: 1,
            type: 'string',
        },
        username: {
            minLength: 1,
            type: 'string',
        },
    },
    required: ['password', 'username'],
    type: 'object',
});
